Core/src/org/sleuthkit/autopsy/report/TableReportSettings.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Autopsy Forensic Browser
+ *
+ * Copyright 2019 Basis Technology Corp.
+ * Contact: carrier <at> sleuthkit <dot> org
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.sleuthkit.autopsy.report;
+
+import java.io.Serializable;
+import java.util.HashMap;
+import java.util.Map;
+import org.sleuthkit.datamodel.BlackboardArtifact;
+
+/**
+ * Class for persisting the selection of the tag types and artifact types used
+ * by the TableReportGenerator class to drive report generation by
+ * TableReportModules.
+ */
+class TableReportSettings implements Serializable  {
+
+    private static final long serialVersionUID = 1L;
+    private Map<BlackboardArtifact.Type, Boolean> artifactTypeSelections = new HashMap<>();
+    private Map<String, Boolean> tagNameSelections = new HashMap<>();
```

Comment:
As for file report settings, I think that we can simplify these settings to Sets by only storing the enabled selections, and that we need some "all" flags. Also, remember that we will need a flag for whether or not to include custom artifact types, since we cannot know what they might be, at least in the elastic computing, command line ingest execution context.

Reply:
I will copy/paste your note into the design document so that it is not lost. I think actual work on this should be done as part of the ReportGeneration refactoring story. I'm trying to make only minimal necessary code changes at this point to keep everything working "as before".